src/mixer.rs

```diff
@@ -0,0 +1,428 @@
+    }
+
+    pub fn has_playback_volume(&self) -> bool {
+        unsafe { alsa::snd_mixer_selem_has_playback_volume(self.1.handle) > 0 }
+    }
+
+    pub fn has_playback_switch(&self) -> bool {
+        unsafe { alsa::snd_mixer_selem_has_playback_switch(self.1.handle) > 0 }
+    }
+
+    pub fn can_capture(&self) -> bool {
+        self.has_capture_volume() || self.has_capture_switch()
+    }
+
+    pub fn can_playback(&self) -> bool {
+        self.has_playback_volume() || self.has_playback_switch()
+    }
+
+    pub fn has_volume(&self) -> bool {
+        self.has_capture_volume() || self.has_playback_volume()
+    }
+
+    /// returns range for capture volume in an array of [min,max] values
+    pub fn get_capture_volume_range(&self) -> [i64;2] {
+        let mut min: i64 = 0;
+        let mut max: i64 = 0;
+        unsafe { alsa::snd_mixer_selem_get_capture_volume_range(self.1.handle, &mut min, &mut max) };
+        [min, max]
+    }
+
+    /// returns array of [min,max] values in decibels*100. To get correct dB value, devide by 100, i.e.
+    ///
+    /// # Example
+    /// ```ignore
+    /// let db_value = selem.capture_decibel_range() as f32 / 100.0;
+    /// ```
+    pub fn get_capture_decibel_range(&self) -> [i64;2] {
+        let mut min: i64 = 0;
+        let mut max: i64 = 0;
+        unsafe { alsa::snd_mixer_selem_get_capture_dB_range(self.1.handle, &mut min, &mut max) };
+        [min, max]
+    }
+
+    /// returns array of [min,max] values
+    pub fn get_playback_volume_range(&self) -> [i64;2] {
+        let mut min: i64 = 0;
+        let mut max: i64 = 0;
+        unsafe { alsa::snd_mixer_selem_get_playback_volume_range(self.1.handle, &mut min, &mut max) };
+        [min, max]
+    }
+
+    /// returns array of [min,max] values in decibels*100. To get correct dB value, devide by 100, i.e.
+    ///
+    /// # Example
+    /// ```ignore
+    /// let db_value = selem.playback_decibel_range() as f32 / 100.0;
+    /// ```
+    pub fn get_playback_decibel_range(&self) -> [i64;2] {
+        let mut min: i64 = 0;
+        let mut max: i64 = 0;
+        unsafe { alsa::snd_mixer_selem_get_playback_dB_range(self.1.handle, &mut min, &mut max) };
+        [min, max]
+    }
+
+    pub fn is_playback_mono(&self) -> bool {
+        unsafe { alsa::snd_mixer_selem_is_playback_mono(self.1.handle) == 1 }
+    }
+
+    pub fn has_capture_channel(&self, channel: i32) -> bool {
+        unsafe { alsa::snd_mixer_selem_has_capture_channel(self.1.handle, channel) > 0 }
+    }
+
+    pub fn has_playback_channel(&self, channel: i32) -> bool {
+        unsafe { alsa::snd_mixer_selem_has_playback_channel(self.1.handle, channel) > 0 }
+    }
+
+    /// Gets name from snd_mixer_selem_channel_name
+    pub fn channel_name(&self, channel: i32) -> Result<String> {
+        let c = unsafe { alsa::snd_mixer_selem_channel_name(channel) };
+        from_const("snd_mixer_selem_channel_name", c).and_then(|s| Ok(s.to_string()))
```

Comment:
What do you propose. Shall we follow alsa naming (although perhaps misnamed) or move this to SelemChannelId?